crates/subspace-networking/src/shared.rs

```diff
@@ -79,3 +85,61 @@ impl Shared {
         }
     }
 }
+
+// Supports exact keys for Kademlia. It uses the Multihash as the canonical type
+// and reduces all the conversions to Multihash conversion.
+// It supports only Sr25519 key pairs for PeerId and Identity multihashes.
+#[derive(Clone, Debug)]
+pub struct IdendityHash(GenericArray<u8, U32>);
+
+impl PreimageIntoKeyBytes<Multihash> for IdendityHash {
+    fn preimage_into_key_bytes(multihash: &Multihash) -> KeyBytes<Self> {
+        const SHA256_HASH_SIZE: usize = 32;
```

Comment:
This is actually a key size. It does match both Sha256 hash size and public key size, but it is incorrect to call it like this here. And please move it to the top of the file.

Reply:
This constant is not intented to change and the scope is limited to this function. I used it to avoid a 'magic number'. Please, confirm you want it at the top of the file. I'll rename it to the `PUBLIC_KEY_SIZE`

Reply:
In that case we already have `PUBLIC_KEY_LENGTH` in `subspace-core-primitives`

Reply:
That's even better. I'll remove it then.